src/receivers/HTTPReceiver.spec.ts

```diff
@@ -521,14 +528,78 @@ describe('HTTPReceiver', function () {
         const fakeRes: ServerResponse = sinon.createStubInstance(ServerResponse) as unknown as ServerResponse;
 
         fakeReq.url = '/test?hello=world';
+        const tempMatch = matchRegex('/test');
+        if (!tempMatch) throw new Error('match failed');
+        const params : ParamsDictionary = tempMatch.params as ParamsDictionary;
 
         fakeReq.method = 'GET';
         receiver.requestListener(fakeReq, fakeRes);
-        assert(customRoutes[0].handler.calledWith(fakeReq, fakeRes));
+        assert(customRoutes[0].handler.calledWith({ ...fakeReq, params }, fakeRes));
 
         fakeReq.method = 'POST';
         receiver.requestListener(fakeReq, fakeRes);
-        assert(customRoutes[0].handler.calledWith(fakeReq, fakeRes));
+        assert(customRoutes[0].handler.calledWith({ ...fakeReq, params }, fakeRes));
+
+        fakeReq.method = 'UNHANDLED_METHOD';
+        assert.throws(() => receiver.requestListener(fakeReq, fakeRes), HTTPReceiverDeferredRequestError);
+      });
+
+      it('should call custom route handler only if request matches route path and method including params', async function () {
+        const HTTPReceiver = await importHTTPReceiver();
+        const customRoutes = [{ path: '/test/:id', method: ['get', 'POST'], handler: sinon.fake() }];
```

Comment:
To verify that the order of these routes is respected by the receiver, we can update these routes to:

```js
const customRoutes = [
  { path: '/test/123', method: ['get', 'POST'], handler: sinon.fake() },
  { path: '/test/:id', method: ['get', 'POST'], handler: sinon.fake() },
];
```

Then check for the following after making a fake request for `/test/123`:

```js
assert(customRoutes[1].handler.notCalled);
```

Another mock request to a different url, say `/test/abc`, could follow the tests for `/test/123` and might include these asserts:

```js
assert(customRoutes[0].handler.notCalled);
assert(customRoutes[1].handler.calledWith({ ...fakeReq, params }, fakeRes));
```

Reply:
I am actually already doing something similar in the next `it` block from your example (line 579) the only difference is that I don't call `assert(customRoutes[1].handler.notCalled);` because I set the response to `sinon.fake.throw('Should not be used.')` as the handler which throws an exception. But I have updated that and I also added a second `it` block to test custom routes in reverse just to make sure that the first path is always called.